src/components/ui/experience.tsx

```
@@ -190,21 +207,47 @@
 
-    if (editingExperience) {
-      setExperienceList((prev) =>
-        prev.map((exp) => (exp.id === editingExperience.id ? { ...experienceFormData } : exp))
-      );
-    } else {
-      setExperienceList((prev) => [...prev, experienceFormData]);
+    setIsLoading(true);
+    setError(null);
+    
+    try {
+      if (editingExperience && editingExperience.id) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)